src/Template/Layout/js/app/app.js

```diff
@@ -306,47 +306,11 @@ const _vueInstance = new Vue({
          * @param {Object} params
          * @returns {String} url
          */
-        buildUrlParams(params) {
-            let url = `${window.location.origin}${window.location.pathname}`;
-            const queryId = '?';
-            const separator = '&';
-            const paramsKeys = Object.keys(params);
-
-            if (paramsKeys && paramsKeys.length) {
-                let fields = [];
-
-                paramsKeys.forEach((key) =>  {
-                    if (params[key]) {
-                        const query = params[key];
-
-                        // parse filter property
-                        if (key === 'filter') {
-                            Object.keys(query).forEach((filterKey) => {
-                                if (typeof query[filterKey] === 'object') {
-                                    const filter = query[filterKey];
-                                    Object.keys(filter).forEach((modifier) => {
-                                        if (filter[modifier] !== '') {
-                                            // look up for param modifier (i.e dates)
-                                            const encoded = encodeURIComponent(filter[modifier]);
-                                            fields.push(`filter[${filterKey}][${modifier}]=${encoded}`);
-                                        }
-                                    });
-                                } else if (query[filterKey] !== '') {
-                                    const encoded = encodeURIComponent(query[filterKey]);
-                                    fields.push(`filter[${filterKey}]=${encoded}`);
-                                }
-                            });
-                        } else {
-                            const encoded = encodeURIComponent(query);
-                            fields.push(`${key}=${encoded}`);
-                        }
-                    }
-                });
-                url += fields.length ? queryId : '';
-                url += fields.join(separator);
-            }
+        buildUrlWithParams(params) {
+            let url = new URL(`${window.location.origin}${window.location.pathname}`);
```

Comment:
```suggestion
            const url = new URL(`${window.location.origin}${window.location.pathname}`);
```
minor, we can use `const` in here